core/iwasm/libraries/lib-wasi-threads/stress_test/mutex_common.h

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef MUTEX_COMMON_H
+#define MUTEX_COMMON_H
+
+#include <pthread.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+typedef struct {
+    pthread_mutex_t *mutex;
+    int *counter;
+    int limit;
+    bool is_sleeping;
+} MutexCounter;
+
+enum constants {
+    NUM_ITER = 100000,
+    NUM_THREADS = 12,
+    NUM_RETRY = 8,
+    RETRY_SLEEP_TIME_US = 1000,
+};
+
+// This enum defines whether thread should sleep to increase contention
+enum sleep_state {
+    NON_SLEEP = 0,
+    SLEEP = 1,
+};
+
+void
+spawn_thread(pthread_t *tid, void *func, void *arg)
+{
+    int status_code = -1;
+    int timeout_us = RETRY_SLEEP_TIME_US;
+    for (int tries = 0; status_code != 0 && tries < NUM_RETRY; ++tries) {
+        status_code = pthread_create(tid, NULL, (void *(*)(void *))func, arg);
+        assert(status_code == 0 || status_code == EAGAIN);
+        if (status_code == EAGAIN) {
+            usleep(timeout_us);
+            timeout_us *= 2;
+        }
+    }
+
+    assert(status_code == 0 && "Thread creation should succeed");
+}
+
+// We're counting how many times each thread was called using this array
+// Main thread is also counted here so we need to make arrays bigger
+typedef struct {
+    int tids[NUM_THREADS + 1];
+    int calls[NUM_THREADS + 1];
+} StatCollector;
+
+StatCollector call_stat;
+
+void
+clear_stat()
+{
+    for (int i = 0; i < NUM_THREADS + 1; ++i) {
```

Comment:
this can be a bit simpler (and more generic, if we add more fields):  `memset(&call_stat, 0, sizeof(call_stat))`

Reply:
Refactored so it's now local.